main.go

```diff
@@ -115,16 +123,49 @@ func main() {
 		os.Exit(1)
 	}
 
+	factory := conditions.InClusterFactory{Client: mgr.GetClient()}
+	namespacedName, err := factory.GetNamespacedName()
+	if err != nil {
+		setupLog.Error(err, "unable to get condition name")
+		os.Exit(1)
+	}
+	condition, err := factory.NewCondition(operatorv2.ConditionType(operatorv2.Upgradeable))
+	if err != nil {
+		setupLog.Error(err, "unable to get OperatorCondition")
+		os.Exit(1)
+	}
 	storageSystemReconciler := &controllers.StorageSystemReconciler{
-		Client:   mgr.GetClient(),
-		Log:      ctrl.Log.WithName("controllers").WithName("StorageSystem"),
-		Scheme:   mgr.GetScheme(),
-		Recorder: controllers.NewEventReporter(mgr.GetEventRecorderFor("StorageSystem controller")),
+		Client:            mgr.GetClient(),
+		Log:               ctrl.Log.WithName("controllers").WithName("StorageSystem"),
+		Scheme:            mgr.GetScheme(),
+		Recorder:          controllers.NewEventReporter(mgr.GetEventRecorderFor("StorageSystem controller")),
+		ConditionName:     namespacedName.Name,
+		OperatorCondition: condition,
 	}
 	if err = storageSystemReconciler.SetupWithManager(mgr); err != nil {
 		setupLog.Error(err, "unable to create controller", "controller", "StorageSystem")
 		os.Exit(1)
 	}
+	apiClient, err := apiclient.New(mgr.GetConfig(), apiclient.Options{
+		Scheme: mgr.GetScheme(),
+	})
+	if err != nil {
+		setupLog.Error(err, "Unable to get Client")
+		os.Exit(1)
+	}
+	condition, err = conditions.InClusterFactory{Client: apiClient}.NewCondition(operatorv2.ConditionType(operatorv2.Upgradeable))
+	if err != nil {
+		setupLog.Error(err, "Unable to get OperatorCondition")
+		os.Exit(1)
+	}
+	err = wait.ExponentialBackoff(retry.DefaultRetry, func() (bool, error) {
+		err = condition.Set(context.TODO(), metav1.ConditionTrue, conditions.WithMessage("Operator is ready"), conditions.WithReason("Ready"))
+		return err == nil, err
+	})
+	if err != nil {
+		setupLog.Error(err, "Unable to update OperatorCondition")
+		os.Exit(1)
+	}
```

Comment:
I think we can ignore this, I really don't see a need for this. Also, it will always set the condition ready as soon as the operator is restarted which should not be the case.

Reply:
> I really don't see a need for this
- It's required https://olm.operatorframework.io/docs/advanced-tasks/communicating-operator-conditions-to-olm/#setting-defaults

> it will always set the condition ready
- Ya, I also thought about it while lifting the code from ocs-op
- It can't be upgradeability false as we need to send a reconcile event for it to change, on a restarted operator this isn't a problem, however on a fresh cluster if none of our watches (wherever we are moving the operatorcond logic) are triggered the cond will not the reconciled
- So, I prefer not to change this and open to be corrected!

Reply:
changed.